pycaret/internal/pycaret_experiment/time_series_experiment.py

```diff
@@ -948,6 +968,26 @@ def setup(
             profile_kwargs=profile_kwargs,
         )
 
+    def _set_default_fig_kwargs(self):
+        """Set the default values for `fig_kwargs` if these are not provided by the user."""
+
+        # `big_data_threshold`: Number of data points above which the hovering for
+        # some plots is disabled. This is needed else the notebooks become very slow.
+        defaults = {
+            "big_data_threshold": 200,
+            "hoverinfo": None,
+            "renderer": None,
+            "template": "ggplot2",
+            "rows": None,
+            "cols": 4,
+            "width": None,
+            "height": None,
+        }
+
+        # Set to default if missing ----
+        for key in defaults:
+            self.fig_kwargs[key] = self.fig_kwargs.get(key, defaults[key])
```

Comment:
you can probably use `dict.setdefault` here

Reply:
Ok. I will try to make an update in the next PR.